src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4412,6 +4412,7 @@ private BoundExpression MakeBadExpressionForObjectCreation(SyntaxNode node, Type
                 var boundInitializer = BindInitializerExpression(syntax: initializerOpt,
                                                                  type: type,
                                                                  typeSyntax: typeSyntax,
+                                                                 isNew: true,
```

Comment:
Is this to suppress cascading errors?

Reply:
This is to differentiate `new C() { $$.P = ... }` and `new C() { P = { $$.P2 = ... }`. The first one is a new instance. The latter is updating an existing instance.

---
In reply to: [554110842](https://github.com/dotnet/roslyn/pull/50200#discussion_r554110842) [](ancestors = 554110842)

Reply:
Sorry, I didn't see the comment was in the handling for bad cases. Yes, for error scenarios, it doesn't seem useful to add other errors.

---
In reply to: [554174635](https://github.com/dotnet/roslyn/pull/50200#discussion_r554174635) [](ancestors = 554174635,554110842)